s3fs/core.py

```diff
@@ -235,6 +235,12 @@ class S3FileSystem(AsyncFileSystem):
     session : aiobotocore AioSession object to be used for all connections.
          This session will be used inplace of creating a new session inside S3FileSystem.
          For example: aiobotocore.session.AioSession(profile='test_user')
+    max_concurrency : int (None)
```

Comment:
I'm fine with any naming convention here, but for reference this is implemented in adlfs and it was named `max_concurrency` there because that's how it's named in the Azure SDK

Reply:
You mean adlfs, or something else?

Reply:
No, I mean it's named `max_concurrency` in the microsoft azure python sdk, which is used under the hood in adlfs (the azure equivalent to boto/botocore)
https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-tune-upload-download-python#set-transfer-options-for-uploads

When support for this was added in adlfs, `max_concurrency` was reused for the parameter name to be consistent with the azure sdk

Reply:
should also note that boto3/s3transfer `TransferConfig` also use `max_concurrency` https://boto3.amazonaws.com/v1/documentation/api/latest/reference/customizations/s3.html#boto3.s3.transfer.TransferConfig

Reply:
OK, we can keep the name, but I'll put in a proposed change to the docstring